src/devices/Ssd13xx/Ssd13xx.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.Ssd13xx.Commands;
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Ssd13xx
+{
+    public abstract class Ssd13xx : IDisposable
+    {
+        protected I2cDevice _i2cDevice;
+
+        public Ssd13xx(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+        }
+
+        /// <summary>
+        /// Verifies value is within a specific range.
+        /// </summary>
+        /// <param name="value">Value to check.</param>
+        /// <param name="start">Starting value of range.</param>
+        /// <param name="end">Ending value of range.</param>
+        /// <returns>Determines if value is within range.</returns>
+        internal static bool InRange(uint value, uint start, uint end)
+        {
+            return (value - start) <= (end - start);
+        }
+
+        /// <summary>
+        /// Send a command to the display controller.
+        /// </summary>
+        /// <param name="command">The command to send to the display controller.</param>
+        protected abstract void SendCommand(ICommand command);
+
+        public abstract void SendCommand(ISharedCommand command);
+
+        /// <summary>
+        /// Send data to the display controller.
+        /// </summary>
+        /// <param name="data">The data to send to the display controller.</param>
+        public virtual void SendData(byte[] data)
+        {
+            const int StackThreshold = 512;
+
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            Span<byte> writeBuffer = data.Length < StackThreshold ?
```

Comment:
if the most usages will use 48*96 buffer I'd just change it to always alloc and write a comment why.
Other option is to perhaps have an internal buffer and expose an API to get that buffer (or Span to that buffer) - this way you don't need to allocate

Reply:
or perhaps we should add a way of passing multiple spans to I2c.Write...